junit-vintage-engine/src/main/java/org/junit/vintage/engine/descriptor/DescriptionUtils.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.vintage.engine.descriptor;
+
+import org.apiguardian.api.API;
+import org.junit.runner.Description;
+
+@API(status = API.Status.INTERNAL, since = "5.8")
+public class DescriptionUtils {
+
+	private DescriptionUtils() {
+	}
+
+	public static String getMethodName(Description description) {
```

Comment:
I saw in the commit comment that this algorithm avoids the use of a RegEx if the display name _looks right_ (or something to that effect).

How confident are you that this algorithm works well for custom JUnit 4 runners?

Reply:
If the regex works for custom runners (it's hardcoded in `Description`), so should this custom algorithm.

Reply:
OK.